### ISSUE TITLE
[Forms] Add 'aria-invalid' to a form entry when it contains constraint violations

### ISSUE THREAD

**Issue by o-alquimista** [symfony/symfony#33116]
When the form entry data violates a constraint, it is necessary to pass the `aria-invalid` attribute to that entry, so screen reader users will know it is incorrect.

Currently, it does not. Here's an example:

The form builder:
```php
$builder
    ->add('title', TextType::class, [
        'label' => 'label.title',
        'help' => 'form_help.title',
        'attr' => [
            'autofocus' => true,
            'aria-required' => 'true',
        ],
    ])
    ->add('summary', TextareaType::class, [
        'label' => 'label.summary',
        'attr' => [
            'rows' => 3,
            'aria-required' => 'true',
        ],
    ])
    ->add('body', TextareaType::class, [
        'label' => 'label.body',
        'attr' => [
            'rows' => 12,
            'aria-required' => 'true',
        ],
    ])
;
```

The form template:
```twig
{{ form_start(form) }}
  {{ form_errors(form) }}

  {{ form_row(form.title) }}
  {{ form_row(form.summary) }}
  {{ form_row(form.body) }}

  <div class="form-group">
    <input type="submit" class="btn btn-info" value="{{ 'label.publish'|trans }}"/>
  </div>
{{ form_end(form) }}
```

The result:
```
<input type="text" id="post_title" name="post[title]" required="required" autofocus="autofocus" aria-required="true" aria-describedby="post_title_help" class="form-control is-invalid" value="a">
```

The Forms component includes the class `.is-invalid` programatically, so I suppose adding `aria-invalid` along with it is possible. It makes the form more accessible by default, requiring less effort from developers.

**Comment by xabbuh** [comment]
Can the `aria-required` attribute be set for all forms that have failed validation constraints? The name makes me think this is only applicable to forms that are required.

**Comment by o-alquimista** [comment]
Actually, now I think the current design of the form keeps error messages in the label so that ARIA attributes are not necessary. The screen reader will read the label and the error messages together. I made some tests with NVDA just now. It checks out.

Adding `aria-invalid` makes the screen reader speak that the form is invalid after it finishes reading the label, but the user will already know it contains errors at that point. `aria-required` for required fields seems like a more useful addition.

**Comment by ismail1432** [comment]
I think that both ideas are good, add aria-required on field with constraint and add aria-invalid when there is a constraint violation
